src/OpenTracing/SpanOptions.php

```diff
@@ -43,7 +43,15 @@ public static function create(array $options)
 
                 case 'tags':
                     foreach ($value as $tag => $tagValue) {
-                        $spanOptions->tags[] = Tag::create($tag, $tagValue);
+                        if ($tag !== (string) $tag) {
+                            throw InvalidSpanOption::invalidTag($tag);
+                        }
+
+                        if (is_array($tagValue) || (!is_scalar($tagValue) && method_exists($tagValue, '__toString'))) {
```

Comment:
I am not sure:

- If `$tagValue` is an `object` with `__toString` then:
`(false || (true && false)) = false // does not throw the exception`
- If `$tagValue` is an `object` without `__toString` then:
`(false || (true && true)) = true // throws the exception`

What I am worried about is the [Resource](http://php.net/manual/en/language.types.resource.php)